server/etcdserver/server.go

```diff
@@ -2181,13 +2195,7 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 		return
 	}
 
-	// keep some in memory log entries for slow followers.
-	compacti := uint64(1)
-	if snapi > s.Cfg.SnapshotCatchUpEntries {
-		compacti = snapi - s.Cfg.SnapshotCatchUpEntries
-	}
-
-	err = s.r.raftStorage.Compact(compacti)
+	err := s.r.raftStorage.Compact(compacti)
```

Comment:
To avoid calling Compact multiple times with the same value, maybe we can add `compacti` field to `etcdProgress`? It's default value can be taken from `s.r.raftStorage.FirstIndex() - 1`.

It would give us a clearer view on raft log compaction.

Reply:
@serathius This approach is more efficient. I’ve implemented your suggestion and added a test case.

Reply:
Awesome work.